chain/network/src/peer_manager/peer_store/mod.rs

```diff
@@ -44,28 +44,192 @@ impl VerifiedPeer {
     }
 }
 
+#[derive(Clone)]
+pub struct Config {
+    pub blacklist: blacklist::Blacklist,
+    /// Nodes will not accept or try to establish connection to such peers.
+    pub boot_nodes: Vec<PeerInfo>,
+    /// If true - connect only to the bootnodes.
+    pub connect_only_to_boot_nodes: bool,
+    /// Remove expired peers.
+    pub peer_expiration_duration: time::Duration,
+    /// Duration of the ban for misbehaving peers.
+    pub ban_window: time::Duration,
+}
+
 /// Known peers store, maintaining cache of known peers and connection to storage to save/load them.
-pub struct PeerStore {
+struct Inner {
+    config: Config,
     store: store::Store,
+    boot_nodes: HashSet<PeerId>,
     peer_states: HashMap<PeerId, KnownPeerState>,
     // This is a reverse index, from physical address to peer_id
     // It can happens that some peers don't have known address, so
     // they will not be present in this list, otherwise they will be present.
     addr_peers: HashMap<SocketAddr, VerifiedPeer>,
-    blacklist: Blacklist,
-    boot_nodes: HashSet<PeerId>,
-    connect_only_to_boot_nodes: bool,
 }
 
+impl Inner {
+    /// Adds a peer which proved to have secret key associated with the ID.
+    ///
+    /// The host have sent us a message signed with a secret key corresponding
+    /// to the peer ID thus we can be sure that they control the secret key.
+    ///
+    /// See also [`Self::add_indirect_peers`] and [`Self::add_direct_peer`].
+    fn add_signed_peer(&mut self, clock: &time::Clock, peer_info: PeerInfo) -> anyhow::Result<()> {
+        self.add_peer(clock, peer_info, TrustLevel::Signed)
+    }
+
+    /// Adds a peer into the store with given trust level.
+    #[inline(always)]
```

Comment:
Why `inline`?

Reply:
Removed. I must have copied it over blindly from somewhere.